sound/soc/codecs/rt711.c

```diff
@@ -457,17 +457,21 @@ static int rt711_set_jack_detect(struct snd_soc_component *component,
 	struct snd_soc_jack *hs_jack, void *data)
 {
 	struct rt711_priv *rt711 = snd_soc_component_get_drvdata(component);
+	int ret;
 
-	rt711->hs_jack = hs_jack;
-
-	if (!rt711->hw_init) {
-		dev_dbg(&rt711->slave->dev,
-			"%s hw_init not ready yet\n", __func__);
-		return 0;
+	ret = pm_runtime_resume_and_get(component->dev);
```

Comment:
I am thinking maybe we only need to get when hs_jack != NULL. The reason that we need to access the register is we want to get the jack's status. But when hs_jack == NULL, we will disable the jack detection feature and we can write the register to cache. Then codec driver will sync it in the device resume callback.

Reply:
We should not have multiple places where the state of the cache is modified, this should only happen on suspend and resume.

We *can* disable jack detection when the card is removed through, that's different to changing the cache settings.